src/H5Dint.c

```diff
@@ -3223,6 +3223,12 @@ H5D__flush(H5D_t *dset, hid_t dset_id)
     HDassert(dset);
     HDassert(dset->shared);
 
+    /* Currently, H5Oflush causes H5Fclose to trigger an assertion failure in metadata cache.
+     * Leave this situation for the future solution */
+    if (H5FD_MPIO == H5F_get_driver_id(dset->oloc.file))
+        HGOTO_ERROR(H5E_DATASET, H5E_CANTFLUSH, FAIL,
+                    "H5Oflush doesn't work properly with MPIO driver for the current design")
```

Comment:
I feel like the HGOTO_ERROR should be protected by an ifdef instead of checking for the MPI-I/O VFD. There may be other parallel VFDs in the future (subfiling, MPI POSIX may return).

Reply:
I changed the use of `H5F_get_driver_id `to `H5F_HAS_FEATURE `.  Running with one rank doesn't make `H5Oflush` succeed.  I tested it in the beginning.